python/pyarrow/pandas_compat.py

```diff
@@ -623,7 +623,21 @@ def _can_definitely_zero_copy(arr):
     metadata.update(pandas_metadata)
     schema = schema.with_metadata(metadata)
 
-    return arrays, schema
+    # If dataframe is empty but with RangeIndex ->
+    # remember the length of the indexes
+    n_rows = None
+    if len(arrays) == 0 and schema is not None:
+        try:
+            kind = index_descriptors[0]["kind"]
+            if kind == "range":
+                start = index_descriptors[0]["start"]
+                stop = index_descriptors[0]["stop"]
+                step = index_descriptors[0]["step"]
+                n_rows = (stop - start - 1)//step + 1
```

Comment:
Is the `- 1` part needed? (in any case for the test example of (0, 10, 3) it's not needed)

And I am just thinking: a `step` can be negative, so we should take that into account (or at least test, it might already work because `stop - start` will then also be negative, so the result would still be positive).

Reply:
Actually, even simpler is to do `n_rows = len(range(start, stop, step))` and let python calculate the length of the range for us!

Reply:
I am just sorry I haven't found this myself :)